test/index.js

```diff
@@ -977,6 +977,24 @@ describe('def', function() {
     eq(dec(new Number(-1)), -2);
   });
 
+  it('supports the "RegexFlags" type', function() {
+    eq($.RegexFlags.name, 'sanctuary-def/RegexFlags');
+    eq($.RegexFlags.test(''), true);
+    eq($.RegexFlags.test('g'), true);
+    eq($.RegexFlags.test('i'), true);
+    eq($.RegexFlags.test('m'), true);
+    eq($.RegexFlags.test('gi'), true);
+    eq($.RegexFlags.test('gm'), true);
+    eq($.RegexFlags.test('im'), true);
+    eq($.RegexFlags.test('gim'), true);
+    //  String objects are not acceptable.
+    eq($.RegexFlags.test(new String('')), false);
+    //  Flags must be alphabetically ordered.
+    eq($.RegexFlags.test('mg'), false);
```

Comment:
Hmm, that's a fair point.

I do tolerate—and often appreciate—looseness in the tools I use, whereas I appreciate strictness in languages and libraries. I appreciate argument juggling when using `ack`, for example, but rail against it in JavaScript libraries.

Another option is to export `G = 1, I = 2, M = 4`, and define `RegexFlags` as `{0, 1, 2, 3, 4, 5, 6, 7}`. This would avoid the order problem as one could write `G | M` or `M | G`. But where would `G`, `I`, and `M` be defined? Perhaps `S.GLOBAL`, `S.IGNORECASE`, and `S.MULTILINE`, leading to `S.GLOBAL | S.MULTILINE`. This is perhaps slightly clearer—but much less convenient—than `'gm'`.

Hypothetical question: Were we to define `S.G`, `S.I`<sup>†</sup>, `S.M`, `S.GI`, etc., would you expect us to define both `S.GM` and `S.MG`?

I'd love to hear from others. Kevin's concern is legitimate, and I'm not convinced that my current position is optimal.

---

† Ignore the fact that `S.I` already exists. This is just a though exercise. :)